src/Interpreters/InterpreterShowColumnsQuery.cpp

```diff
@@ -93,6 +99,74 @@ String InterpreterShowColumnsQuery::getRewrittenQuery()
 
 }
 
+String InterpreterShowColumnsQuery::getMySQLQuery()
+{
+    WriteBufferFromOwnString mysql_specific_query;
+
+    mysql_specific_query << "SELECT name AS field, "
+        << "CASE "
+        << "  WHEN startsWith(type, 'Nullable') THEN "
+        << "    CASE "
+        << "      WHEN substring(type, 10, length(type) - 10) IN ('UInt8', 'Int8') THEN 'tinyint' "
+        << "      WHEN substring(type, 10, length(type) - 10) IN ('UInt16', 'Int16') THEN 'smallint' "
```

Comment:
This introduces an implicit dependency between this file and every file inside the DataTypes directory.

Reply:
This is a good point. Perhaps we should instead build this string by iterating over the TypeIndex enum for all of the ClickHouse types and get the equivalent MySQL type with a function getSQLCompatibleName in IDataType (as you mentioned before), then look it up in a type equivalency map. That should solve the implicit dependency and make this more manageable although we would need to maintain the equivalency map.